File: backend/src/main/java/br/gulli/spring/service/JobService.java
package br.gulli.spring.service;

import java.util.List;

import br.gulli.spring.domain.Job;
import br.gulli.spring.dto.JobDTO;

public interface JobService {

	List<Job> findAllJobs();
	
	JobDTO cadastrar(JobDTO dto) throws Exception;
	
}

File: frontend/src/app/core/job/job.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { FiltroDTO } from '../models/filtroDTO';
import { UserService } from '../services/user.service';
import { JobDTO } from './job-dto';

const URL = '/api/v2/jobs';

@Injectable()
export class JobService {

    httpOptions = {
        headers: new HttpHeaders({
            'Content-Type': 'application/json',
            'Authorization': 'Bearer ' + this.userService.accessToken
        })
    };

    constructor(
        private userService: UserService,
        private httpClient: HttpClient
    ) { }

    getAllJobs() {
        return this.httpClient.get(URL, this.httpOptions);
    }


    cadastrar(filtro: FiltroDTO) {
        return this.httpClient.post<JobDTO>(URL, filtro, this.httpOptions);
    }

}